### ISSUE TITLE
(possibly) unneeded/unused `resource null_resource.module_depends_on` created

### ISSUE THREAD

**Issue by swapzero** [terraform-google-modules/terraform-google-kubernetes-engine#1194]
### TL;DR

A _possibly_ unneeded/unused resource is created when calling the module and _it's the only one_ created as part of that module call (`enabled = false` because defaults are used for parameters that affect `enabled` value) 

Resource:
```tf

$ terraform state show ...

resource "null_resource" "module_depends_on" {
    id       = "4730592283587992509"
    triggers = {
        "value" = "2"
    }
}
```
The code that creates the resource is [here](https://github.com/terraform-google-modules/terraform-google-gcloud/blob/v3.1.1/main.tf#L62-L68) and it gets called when module [gcloud_delete_default_kube_dns_configmap](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-private-cluster-update-variant/dns.tf#L22-L40) is called.

Apparently, it is the only resource that doesn't use `enabled` module option, therefore it gets created even if `enabled` is `false` and occasionally shows changes when the nodepools are added/removed.

### Expected behavior

The resource should also use `enabled` flag to avoid creation (module options that affect `enabled` are on defaults so in my case `enabled = false`)

```diff
 resource "null_resource" "module_depends_on" {
-  count = length(var.module_depends_on) > 0 ? 1 : 0
+  count = (length(var.module_depends_on) > 0 && var.enabled) ? 1 : 0
  triggers = {
    value = length(var.module_depends_on)
  }
}
```

### Observed behavior

The only resource that is created as part of the submodule call and it has no effect alone.

### Terraform Configuration

```hcl
module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant"
  version = "20.0.0"
...
}
```

### Terraform Version

```sh
0.15.5
```

**Comment by morgante** [comment]
Since the resource has no impact, I don't think this is a major issue either way.
